Documentation/install/upgrade.rst

```diff
@@ -278,6 +278,35 @@ Changes that may require action
  * The ``--serve`` option was removed from cilium-bugtool in favor of a much
    reduced binary size. If you want to continue using the option, please use an
    older version of the cilium-bugtool binary.
+ * The always-on :ref:`DNS Polling` used by ``toFQDNs.matchName`` rules is
+   disabled by default in 1.4.x. Pass ``--tofqdns-enable-poller`` to
+   cilium-agent to enable it. It has been replaced by :ref:`DNS Proxy` support
+   but this requires policy changes. The :ref:`DNS Proxy` provides DNS IP
+   information to the toFQDNs rules, similar to :ref:`DNS Polling`, but
+   restricted to actual DNS traffic seen by Endpoints.
+
+   It is possible to prepare policy rules to use the :ref:`DNS Proxy` before an
+   upgrade to 1.4. Then newly added ``toFQDNs.matchPattern`` and L7
+   ``toPorts.rules.dns.matchName/matchPattern`` fields will be ignored by older
+   cilium versions and can be in-place prior to an upgrade.
+
+   In the following example egress rules to allow DNS access to kube-dns in
+   added, including a L7 DNS rule to allow interception of DNS traffic without
+   blocking any. The ``toFQDNs`` rules do not need to be modified. 
```

Comment:
I had a few thoughts when reading this:
* Do I want to enable dns-poller?
* Why is it disabled by default now?
* Some of the wording is a bit too passive

I've tried to streamline the description above, I hope this is an improvement. Feel free to take/leave whichever parts of the below that you think make positive changes to the wording:

```
 * The :ref:`DNS Polling` option used by ``toFQDNs.matchName`` rules is         
   disabled by default in 1.4.x due to :ref:`limitations in the implementation  
   <DNS Polling>`. It has been replaced by :ref:`DNS Proxy` support, which must 
   be explicitly enabled via changes to the policy described below. To ease     
   upgrade, users may opt to enable the :ref:`DNS Polling` in v1.4.x by adding  
   the ``--tofqdns-enable-poller`` option to cilium-agent without changing      
   policies.                                                                    
                                                                                
   Policy rules may be prepared to use the :ref:`DNS Proxy` before an upgrade   
   to 1.4. The new policy rule fields ``toFQDNs.matchPattern`` and              
   ``toPorts.rules.dns.matchName/matchPattern`` will be ignored by older cilium 
   versions and can be safely implemented prior to an upgrade.                  
                                                                                
   The following example allows DNS access to ``kube-dns`` via the :ref:`DNS    
   Proxy` and allows all DNS requests to ``kube-dns``. For completeness,        
   ``toFQDNs`` rules are included for examples of the syntax for those L3       
   policies as well. Existing ``toFQDNs`` rules do not need to be modified.  
```